### DISCUSSION TITLE
Merge to main branch only one commit

### DISCUSSION THREAD

**Discussion by N1C0exe** [community/community#37035] [Actions] [answered]
Hi is possible into github repo to Merge into the main branch only one commit made ?

**Comment by SilasRodrigues19** [comment]
@Nico-ITA Hi.

You can create one branch with the following command
``` 
git checkout -b develop
```
This command will create a branch called ```develop``` and switch to it. Then you can commit your changes and after use the following command

```
git push origin develop
```

After all that, if you access your repository, an option will appear to open PR and merge into the main branch.

**Reply by N1C0exe** [reply]
yes but maybe I want to merge only a specific commit made . Not all of them . From github site into PR page this cant be odne afaik

**Reply by SilasRodrigues19** [reply]
When I want to do this I commit the specific file I want to merge:

```git add <file>``` or ```git add <directory>```

Then I commit the changes

``` git commit -m "Commit message" ``` and ```git push origin branch-name```

**Comment by N1C0exe** [comment]
so I need to to this via console not via web interface ?

**Reply by SilasRodrigues19** [reply]
Yeah

**Comment by ossanna16** [comment, marked as answer]
Hi there @Nico-ITA and welcome to our community 😄. Thanks so much for asking a great question!
